cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/PreventBrokerScaleDownOperations.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.operator.cluster.model.KafkaCluster;
+import io.strimzi.operator.common.AdminClientProvider;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.operator.resource.SecretOperator;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListTopicsOptions;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.TopicPartitionInfo;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.HashSet;
+/**
+ * Class which contains several utility function which check if broker scale down can be done or not.
+ */
+public class PreventBrokerScaleDownOperations {
```

Comment:
Not so critical. But would `PreventBrokerScaleDownCheck` be a better name? It does not have any _operations_. It just checks the topics.